### PR TITLE
Maven plugin 2.0 test

### PR DESCRIPTION
**Pull request by josephca**
- Merged 2 modules into one to adopt maven liberty plugin 2.0

### CODE REVIEW

File: docs/Using-WDT.md [hunk 1]
```diff
@@ -63,59 +63,35 @@ This assumes you have the Gradle [Buildship](https://projects.eclipse.org/projec
...
-
-* Note: Some versions of WDT incorrectly map the cdi-1.2 dependency to the CDI 1.0 Facet, which prevents the *Run As ...* operation in step 2 from succeeding. If this happens, Right-click on the `async-servletnio-application` project, and select *Properties*, then select *Project Facets* in the left-hand pane. Change the the "Context and dependency injection (CDI)" facet to use version 1.2, at which point, step 2 (above) should work.
+2. To run the server with the Servlet sample execute:
+    ```bash
+    $ mvn liberty:run-server
+    ```
 
+3. Confirm web browser opens on "http://localhost:9082/servletnio/" with 5 hyperlinks to run samples
 
 #### Tips
```
Comment (patricktiu) [inline docs/Using-WDT.md]: Does this Tips still apply after the project structure changed?

File: docs/Using-WDT.md [hunk 2]
```diff
@@ -63,59 +63,35 @@ This assumes you have the Gradle [Buildship](https://projects.eclipse.org/projec
 2.  Right-click on this folder, and select *Copy path to Clipboard*
 3.  Select menu *File -> Import -> Maven -> Existing Maven Projects*
 4.  In the Root Directory textbox, Paste in the repository directory.
-5.  Select *Browse...* button and select *Finish* (confirm it finds 3 pom.xml files)
-6.  This will create 3 projects in Eclipse: async-servlet-nio, async-servlet-nio-application, and async-servlet-nio-wlpcfg
+5.  Select *Browse...* button and select *Finish*
```
Comment (patricktiu) [inline docs/Using-WDT.md]: This project doesn't build with Gradle.  Should we remove the `Building with Gradle` section?

File: docs/Using-WDT.md [hunk 3]
```diff
@@ -63,59 +63,35 @@ This assumes you have the Gradle [Buildship](https://projects.eclipse.org/projec
...
-###### Run Maven install
+###### Running in Eclipse with Maven
 
-1. Right-click on async-servlet-nio/pom.xml
-2. *Run As > Maven build...*
-3. In the *Goals* section enter "install"
-4. Click *Run*
+1. Right-click the project and select **Run As > Maven Clean**.
+2. Right-click the project and select **Run As > Maven Install**.
+3. Right-click the project and select **Run As > Maven build...** then add 'liberty:run-server' in Goals.
```
Comment (patricktiu) [inline docs/Using-WDT.md]: why not use Run as > Run on Server to start the server after import the project into WDT.

File: docs/Using-WDT.md (hunk 2 above)
Comment (josephca) [inline docs/Using-WDT.md]: Will remove the Gradle Sections.

File: docs/Using-WDT.md (hunk 1 above)
Comment (josephca) [inline docs/Using-WDT.md]: I think we don't need it anymore - will remove it.

File: pom.xml [hunk 4]
```diff
@@ -30,9 +24,174 @@
...
+    <profiles>
+        <profile>
+            <id>usr-package</id>
+            <properties>
+                <packaging.type>usr</packaging.type>
+            </properties>
+        </profile>
+    </profiles>
+    
+    <dependencies>
```
Comment (josephca) [inline pom.xml]: Fixed it.

File: docs/Using-WDT.md (hunk 3 above)
Comment (josephca) [inline docs/Using-WDT.md]: Fixed.

File: pom.xml [hunk 5]
```diff
@@ -30,9 +24,174 @@
...
+                    <assemblyArtifact>
+                        <groupId>com.ibm.websphere.appserver.runtime</groupId>
+                        <artifactId>wlp-javaee7</artifactId>
+                        <version>17.0.0.1</version>
+                        <type>zip</type>
+                    </assemblyArtifact>
+                    <configFile>src/main/liberty/config/server.xml</configFile>
+                    <include>${packaging.type}</include>
+                    <installAppPackages>project</installAppPackages>
+                    <looseApplication>false</looseApplication>
```
Comment (josephca) [inline pom.xml]: Yes, we can -fixed.

File: pom.xml [hunk 6]
```diff
@@ -45,41 +45,35 @@
...
-            <version>7.0</version>
-            <scope>provided</scope>
-        </dependency>
-        <dependency>
-            <groupId>jstl</groupId>
-            <artifactId>jstl</artifactId>
-            <version>1.2</version>
-            <scope>provided</scope>
+            <groupId>org.apache.cxf</groupId>
+            <artifactId>cxf-rt-rs-client</artifactId>
```
Comment (patricktiu) [inline pom.xml]: This dependency `cxf-rt-rs-client` is defined twice.

Comment (josephca) [inline pom.xml]: good catch - fixed.

**Comment by patricktiu** [comment]
Did you try to import the sample into WDT and run as server?  I am getting some error.

**Comment by josephca** [comment]
Yes, I tested with WDT and run as server - no error was found.

**Comment by patricktiu** [comment]
I don't know why your eclipse didn't flag the error.   You have an invalid `package net.wasdev.servlet.nio.it.test;` statement in https://github.com/josephca/sample.async.servletnio/blob/maven-plugin-2.0-test/src/test/java/it/test/ServletIT.java

**Comment by josephca** [comment]
My bad. I overlooked red x. Checked in the fix.

File: docs/Using-cmd-line.md [hunk 7]
```diff
@@ -24,6 +24,12 @@ Use the following steps to run the application with Maven:
     ```bash
     $ mvn clean install
     ```
+    
+    If you want to also run the functional tests then you need to [Download WAS Liberty](/docs/Downloading-WAS-Liberty.md) and pass in the location of your install as the system property libertyRoot:
+
+    ```bash
+    $ mvn -DlibertyRoot=<LibertyInstallLocation> install
```
Comment (patricktiu) [inline docs/Using-cmd-line.md]: It doesn't deploy and run the test against liberty in this location `<LibertyInstallLocation>`

Comment (josephca) [inline docs/Using-cmd-line.md]: Removed the section.

**Pull request merged**